wurst/file/Base64.wurst

```diff
@@ -0,0 +1,284 @@
+    This value has been tuned to work under all optimization settings and
+    with stacktraces included.
+**/
+@configurable public constant ENCODES_PER_ROUND = 1000
+/**
+    Specifies how many chunks to decode per a single `execute()` call.
+    This value has been tuned to work under all optimization settings and
+    with stacktraces included.
+**/
+@configurable public constant DECODES_PER_ROUND = 100
+
+// RFC 4648 compliant Base64 charmap.
+constant CHARMAP = [
+    "A","B","C","D","E","F","G","H","I","J","K","L","M","N","O","P",
+    "Q","R","S","T","U","V","W","X","Y","Z","a","b","c","d","e","f",
+    "g","h","i","j","k","l","m","n","o","p","q","r","s","t","u","v",
+    "w","x","y","z","0","1","2","3","4","5","6","7","8","9","+","/"
+]
+
+// RFC 4648 compliant Base64 reverse charmap.
+constant REVERSE_CHARMAP = [
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 62, 0, 0, 0, 63, 52, 53, 54,
+    55, 56, 57, 58, 59, 60, 61, 0, 0, 0, 0, 0, 0, 0, 0, 1, 2,
+    3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19,
+    20, 21, 22, 23, 24, 25, 0, 0, 0, 0, 0, 0, 26, 27, 28, 29, 30,
+    31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47,
+    48, 49, 50, 51, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
+]
+
+public class Base64Encoder
+    private ChunkedString chunkedString
+
+    // We append only one char each time, and strings in WC3 get fully copied on each concatenation.
+    // To optimize it and lower the times of copying quite big chunks, append by smaller chunks.
+    private var stringBuffer = ""
+    private var chars = 0
+    private static constant MAX_CHARS = 32
+
+    private var buffer = 0
+    private var bytes = 0
+    private static constant MAX_BYTES = 3
+
+    construct()
+        this.chunkedString = new ChunkedString()
+
+    /** This constructor consumes the ByteBuffer. If you want to save it, use the 'write' method. */
+    construct(ByteBuffer data)
+        this.chunkedString = new ChunkedString()
+
+        writeData(data)
+        destroy data
+
+    construct(int maxChunkLength)
+        this.chunkedString = new ChunkedString(maxChunkLength)
+
+    /** This constructor consumes the ByteBuffer. If you want to save it, use the 'write' method. */
+    construct(ByteBuffer data, int maxChunkLength)
+        this.chunkedString = new ChunkedString(maxChunkLength)
+
+        writeData(data)
+        destroy data
+
+    ondestroy
+        if chunkedString != null
```

Comment:
if you always initialize the member anyway, it will be constant and you can get rid of this nullcheck

Reply:
This nullcheck is here because the string is returned to the user in 'intoData' method, so there it's set to null and prevent it from being destroyed.